Kommunicate/Classes/KMConversationListViewController.swift

```diff
@@ -0,0 +1,571 @@
+            NSLog("Reloadtable notification received")
+
+            guard let weakSelf = self, let list = notification.object as? [Any] else { return }
+            weakSelf.viewModel.updateMessageList(messages: list)
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "USER_DETAILS_UPDATE_CALL"), object: nil, queue: nil, using: { [weak self] notification in
+            NSLog("update user detail notification received")
+
+            guard let weakSelf = self, let userId = notification.object as? String else { return }
+            print("update user detail")
+            weakSelf.viewModel.updateUserDetail(userId: userId) { (success) in
+                if success {
+                    weakSelf.tableView.reloadData()
+                }
+            }
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "UPDATE_CHANNEL_NAME"), object: nil, queue: nil, using: { [weak self] _ in
+            NSLog("update group name notification received")
+            guard let weakSelf = self, weakSelf.view.window != nil else { return }
+            print("update group detail")
+            weakSelf.tableView.reloadData()
+        })
+    }
+
+
+     func removeObserver() {
+        if alMqttConversationService != nil {
+            alMqttConversationService.unsubscribeToConversation()
+        }
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "pushNotification"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "newMessageNotification"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "reloadTable"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "USER_DETAILS_UPDATE_CALL"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "UPDATE_CHANNEL_NAME"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: UIResponder.keyboardDidHideNotification, object: nil)
+    }
+
+    open override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        activityIndicator.center = CGPoint(x: view.bounds.size.width / 2, y: view.bounds.size.height / 2)
+        activityIndicator.color = UIColor.gray
+        view.addSubview(activityIndicator)
+        view.bringSubviewToFront(activityIndicator)
+        edgesForExtendedLayout = []
+        viewModel.prepareController(dbService: dbService)
+    }
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        alMqttConversationService = ALMQTTConversationService.sharedInstance()
+        alMqttConversationService.mqttConversationDelegate = self
+        alMqttConversationService.subscribeToConversation()
```

Comment:
We can create a separate method for mqtt setup(and subscription) for better readability. Same for view related setup in `viewWillAppear`.